Packs/XSOAR_Engineer_Training/Integrations/XSOAREngineerTraining/XSOAREngineerTraining.py

```diff
@@ -0,0 +1,369 @@
+        if item["urlCategory"] == "PHISH":
+            item["url"] = f"https://xsoar.pan.dev/{randrange(1,88)}/login.php"
+        if item["urlCategory"] == "SPAM":
+            item["url"] = f"https://xsoar.pan.dev/{randrange(1,88)}/getnewbike"
+
+        data.append(item)
+        count += 1
+    return data
+
+
+def lookup_ad_user(lookup, attribute):
+    """
+    Returns the user details from the USERS global var
+    """
+    found_user = False
+    for u in USERS:
+        if u[attribute] == lookup:
+            found_user = True
+            return u
+            break
+
+    if not found_user:
+        return []
+    return None
+
+
+def create_ad_user_output(user):
+    """
+    returns the user object for using the USER_TEMPLATE
+    """
+    user_template = {
+        "attributes": {
+            "displayName": [
+                f"{user['displayname']}"
+            ],
+            "mail": [
+                f"{user['email']}"
+            ],
+            "manager": [
+                f"{user['manager']}"
+            ],
+            "memberOf": [
+                f"{user['group']}"
+            ],
+            "name": [
+                f"{user['displayname']}"
+            ],
+            "sAMAccountName": [
+                f"{user['samaccountname']}"
+            ],
+            "userAccountControl": [
+                512
+            ],
+            "dn": f"{user['dn']}"
+        },
+        "account": {
+            "DisplayName": [
+                f"{user['displayname']}"
+            ],
+            "Email": [
+                f"{user['email']}"
+            ],
+            "Manager": [
+                f"{user['manager']}"
+            ],
+            "Groups": [
+                f"{user['group']}"
+            ],
+            "Type": "AD",
+            "Username": [
+                f"{user['samaccountname']}"
+            ],
+            "ID": f"{user['dn']}"
+        }
+    }
+
+    return user_template
+
+
+""" Command functions """
+
+
+def get_events_command(args):
+    """
+    returns the events, in this case the mock data
+    """
+    args.get("eventTypes")
+    total = randrange(10, 20)
+    events = mock_data(total)
+    readable = tableToMarkdown("Training Events", events)
+    results = CommandResults(readable_output=readable, ignore_auto_extract=True)
+    return results
+
+
+def simple_response_command(command):
+    """
+    returns a text response for the Active Directory and send mail Commands
+    """
+    command_map = {
+        'ad-expire-password': 'Expired password successfully',
+        'ad-set-new-password': 'User password successfully set',
+        'send-mail': 'XSOAR Engineer Training: fake email notification not sent'
+    }
+    return CommandResults(readable_output=command_map[command], ignore_auto_extract=True)
+
+
+def ad_get_user_command(args):
+    """
+    Returns the user details from our simulated AD if the user is found
+    """
+
+    # error handing for Ori
```

Comment:
Out of curiosity, who is Ori?

Reply:
He was our old CS leader, once told me my error handing sucked (as a joke).  

Removed.